go/tools/codegen/goimports_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package codegen
+
+import (
+	"testing"
+
+	"github.com/dave/jennifer/jen"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_FormatGenFile(t *testing.T) {
+	tests := []struct {
+		name        string
+		file        *jen.File
+		expectedErr string
+	}{
+		{
+			name:        "some-file",
+			file:        jen.NewFile("some-file"),
+			expectedErr: "Error 1:13: expected ';', found '-' while formatting source",
+		},
+		{
+			name:        "random",
+			file:        jen.NewFile("random"),
+			expectedErr: "",
```

Comment:
nit, you can remove this field
```suggestion
```

Reply:
In addition to this, do you think that I should get rid of the loop as well because there's only a single test case.

Reply:
you have two tests, right?

Reply:
Opps, I messed up with the diffs while reading the suggestions. Thanks for letting me know.

Reply:
Fixed it.